docs/proposals/resource-quota.md

```diff
@@ -0,0 +1,404 @@
+  // matchLabels is a map of {key,value} pairs. A single {key,value} in the matchLabels
+  // map is equivalent to an element of matchExpressions, whose key field is "key", the
+  // operator is "In", and the values array contains only "value". The requirements are ANDed.
+  MatchLabels map[string]string `json:"matchLabels,omitempty"`
+  // matchExpressions is a list of label selector requirements. The requirements are ANDed.
+  MatchExpressions []LabelSelectorRequirement `json:"matchExpressions,omitempty"`
+}
+
+// A label selector requirement is a selector that contains values, a key, and an operator that
+// relates the key and values.
+type LabelSelectorRequirement struct {
+  // key is the label key that the selector applies to.
+  Key string `json:"key" patchStrategy:"merge" patchMergeKey:"key"`
+  // operator represents a key's relationship to a set of values.
+  // Valid operators ard In, NotIn, Exists and DoesNotExist.
+  Operator LabelSelectorOperator `json:"operator"`
+  // values is an array of string values. If the operator is In or NotIn,
+  // the values array must be non-empty. If the operator is Exists or DoesNotExist,
+  // the values array must be empty. This array is replaced during a strategic
+  // merge patch.
+  Values []string `json:"values,omitempty"`
+}
+
+// A label selector operator is the set of operators that can be used in a selector requirement.
+type LabelSelectorOperator string
+
+const (
+  LabelSelectorOpIn           LabelSelectorOperator = "In"
+  LabelSelectorOpNotIn        LabelSelectorOperator = "NotIn"
+  LabelSelectorOpExists       LabelSelectorOperator = "Exists"
+  LabelSelectorOpDoesNotExist LabelSelectorOperator = "DoesNotExist"
+)
+```
+
+### FieldSelector
+
+The `v1` API will add a similar construct to `LabelSelector` to support
+storage of `FieldSelector` declarations.  A `FieldSelector` supports
+a `Kind` and `APIVersion` to provide context to which object the
+selector applies against.
+
+```
+// A field selector is a field query over a set of resources. The result of matchFields and
+// matchExpressions are ANDed. An empty field selector matches all objects. A null
+// field selector matches no objects.
+type FieldSelector struct {
+  // Kind of object that scopes the selector; More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#types-kinds"
+  // If omitted, the Kind is inferred from the context
+  Kind string `json:"kind,omitempty"`
+  // API version of object that scopes the selector
+  // If omitted, the API version is inferred from the context
+  APIVersion string `json:"apiVersion,omitempty"`
+  // matchFields is a map of {fieldPath,value} pairs. A single {fieldPath,value} in the matchFields
+  // map is equivalent to an element of matchExpressions, whose fieldPath field is "fieldPath", the
+  // operator is "In", and the values array contains only "value". The requirements are ANDed.
+  MatchFields map[string]string `json:"matchFields,omitempty"`
```

Comment:
If the reason you have this is because you were patterning after LabelSelector, you don't need it -- the only reason it was in LabelSelector was for backward compatibility.